addons/context_actions/functions/fnc_getVehicleWeaponActions.sqf

```diff
@@ -0,0 +1,104 @@
+#include "script_component.hpp"
+/*
+ * Author: Ampersand
+ * Dynamically creates child actions to select a weapon for turrets with multiple weapons/muzzles/magazines.
+ *
+ * Arguments from CfgContext insertChildren:
+ * 0: None <OBJECT>
+ * 1: None <OBJECT>
+ * 2: None <OBJECT>
+ * 3: None <OBJECT>
+ * 4: None <OBJECT>
+ * 5: Vehicle <OBJECT>
+ *
+ * Return Value:
+ * 0: Child Actions <ARRAY>
+ *
+ * Example:
+ * ["", "", "", "", "", _vehicle] call zen_context_actions_fnc_getVehicleWeaponActions
+ *
+ * Public: No
+ */
+
+params ["", "", "", "", "", "_vehicle"];
+
+private _cfgAmmo = configFile >> "CfgAmmo";
+private _cfgMagazines = configFile >> "CfgMagazines";
+private _cfgWeapons = configFile >> "CfgWeapons";
+
+private _currentWeapon = currentWeapon _vehicle;
+private _currentWeaponMuzzles = getArray (_cfgWeapons >> _currentWeapon >> "muzzles");
+private _currentMagazine = currentMagazine _vehicle;
+
+private _primaryGunner = gunner _vehicle;
+private _primaryTurret = [0];
+if (isNull _primaryGunner) then {
+    _primaryGunner = driver _vehicle;
+    _primaryTurret = [-1];
+};
+if (isNull _primaryGunner) exitWith {};
```

Comment:
```suggestion
private _primaryGunner = gunner _vehicle;
private _primaryTurret = [0];

if (isNull _primaryGunner) then {
    _primaryGunner = driver _vehicle;
    _primaryTurret = [-1];
};

if (isNull _primaryGunner) exitWith {};
```

Majority of this code and the next code is shared from addons/context_actions/functions/fnc_canSelectVehicleWeapon.sqf, it'd be nice if it was extracted into it's own function.

Reply:
Runtimes on a tank:
`canSelectVehicleWeapon` 0.0463 ms
`getVehicleWeaponActions` 1.9666 ms
Is it better to keep the quicker condition check function:
`condition = QUOTE(_hoveredEntity call FUNC(canSelectVehicleWeapon));`
or is the difference small enough on a 1-run-per-context-open that it should just depend on the child actions?